build-system/scripts/calculate_content_hash

```diff
@@ -11,9 +11,45 @@ REBUILD_PATTERNS=$(query_manifest rebuildPatterns $REPOSITORY)
 
 AWK_PATTERN=$(echo $REBUILD_PATTERNS | sed 's/ /|/g')
 cd "$(git rev-parse --show-toplevel)"
+
 # an example line is
 #   100644 da9ae2e020ea7fe3505488bbafb39adc7191559b 0       yarn-project/world-state/tsconfig.json
 # this format is beneficial as it grabs the hashes from git efficiently
 # we will next filter by our rebuild patterns
 # then we pipe the hash portion of each file to git hash-object to produce our content hash
-git ls-tree -r $COMMIT_HASH | awk -v pattern="($AWK_PATTERN)" '$4 ~ pattern {print $3}' | git hash-object --stdin
+if [ -z "${CONSIDER_WORKING:-}" ]; then
+  git ls-tree -r $COMMIT_HASH | awk -v pattern="($AWK_PATTERN)" '$4 ~ pattern {print $3}' | git hash-object --stdin
```

Comment:
Can I be the bash complexity police? : D
What about something like...
```
if ! git diff-index --quiet HEAD --; then
    echo "Error: You have unstaged changes, refusing to calculate content hash. Please make an intermediate commit, and amend it as needed."
    exit 1
fi
```
but you're the captain